update Changelog for 1.8.0-alpha03 release (#5231)
JetBrains/compose-multiplatform: CHANGELOG.md

# 1.8.0-alpha03 (February 2025)

_Changes since 1.8.0-alpha02_

## Highlights

### Multiple Platforms

- [Compose Multiplatform codebase is fully migrated to K2](https://github.com/JetBrains/compose-multiplatform-core/pull/1778). Please note that native and web klibs can be consumed only with Kotlin 2.1.0 or newer. Also, due to underlying changes in the compiler plugin, it's better to recompile libraries against the new version. Please let us know if you find any compatibility issues during this migration
- [Implement multiplatform `BackHandler` and `PredictiveBackHandler`](https://github.com/JetBrains/compose-multiplatform-core/pull/1771). And use them in material3 widgets and androidx-navigation library

## Breaking Changes

### Multiple Platforms

> - [IdlingResource](https://developer.android.com/reference/kotlin/androidx/compose/ui/test/IdlingResource) inteface was moved from commonMain to android and desktop source sets. The related experimental methods of [ComposeUiTest](https://developer.android.com/reference/kotlin/androidx/compose/ui/test/ComposeUiTest) were moved too. They are not available for Web and iOS anymore. Consider using [waitUntil function][(https://developer.android.com/reference/kotlin/androidx/compose/ui/test/ComposeUiTest#waitUntil(kotlin.String,kotlin.Long,kotlin.Function0)) as an alternative](https://github.com/JetBrains/compose-multiplatform-core/pull/1822). Note: it's a breaking change only for Web and iOS, but not for Desktop and Android

Reviewer: There is a small typo in "interface"

### iOS

> - [Experimental classes CupertinoScrollDecayAnimationSpec and CupertinoOverscrollEffect are removed from public API](https://github.com/JetBrains/compose-multiplatform-core/pull/1806)

Reviewer 2: ```suggestion
- [Experimental classes `CupertinoScrollDecayAnimationSpec` and `CupertinoOverscrollEffect` are removed from public API](https://github.com/JetBrains/compose-multiplatform-core/pull/1806)
```

## Features

### Multiple Platforms

> - [Adopt a new Clipboard interface with suspend functions, which work correctly on all targets including Web](https://github.com/JetBrains/compose-multiplatform-core/pull/1796). The ClipboardManager was deprecated because it was not possible to correctly implement it for Web

Reviewer 2: ```suggestion
- [Adopt a new `Clipboard` interface with suspend functions, which work correctly on all targets including Web](https://github.com/JetBrains/compose-multiplatform-core/pull/1796). The `ClipboardManager` was deprecated because it was not possible to correctly implement it for Web
```

### iOS

- [Support VoiceControl on iOS](https://github.com/JetBrains/compose-multiplatform-core/pull/1780)

> - [AccessibilitySyncOptions removed](https://github.com/JetBrains/compose-multiplatform-core/pull/1780). The accessibility tree is built on demand

Reviewer 2: ```suggestion
- [`AccessibilitySyncOptions` removed](https://github.com/JetBrains/compose-multiplatform-core/pull/1780). The accessibility tree is built on demand
```

- [Calculate the order and location of semantic elements in the same way as it's done on Android](https://github.com/JetBrains/compose-multiplatform-core/pull/1809)

> - [Support UIAccessibilityContainerTypeSemanticGroup for traversal groups](https://github.com/JetBrains/compose-multiplatform-core/pull/1809)

Reviewer 2: ```suggestion
- [Support `UIAccessibilityContainerTypeSemanticGroup` for traversal groups](https://github.com/JetBrains/compose-multiplatform-core/pull/1809)
```

> - [Compose works correctly with nested UIScrollViews, as well as within UIScrollViews](https://github.com/JetBrains/compose-multiplatform-core/pull/1818)

Reviewer 2: ```suggestion
- [Compose works correctly with nested `UIScrollView`s, as well as within `UIScrollView`s](https://github.com/JetBrains/compose-multiplatform-core/pull/1818)
```

- [Fixes an issue where it's not possible to close the UIMenu that appears over the Compose content](https://github.com/JetBrains/compose-multiplatform-core/pull/1818)
- [Fix touches tracking for multitouch gestures](https://github.com/JetBrains/compose-multiplatform-core/pull/1827)

### Desktop

- [Elements marked with `Modifier.semantics { hideFromAccessibility() }` should now be correctly hidden from a11y](https://github.com/JetBrains/compose-multiplatform-core/pull/1798)
- [Fixed the positioning of the IME candidate popup for `TextField(TextFieldState)`](https://github.com/JetBrains/compose-multiplatform-core/pull/1794) (aka `BasicTextField2`)
- [Fixed input of diacritics via long-press on macOS in `TextField(TextFieldState)`](https://github.com/JetBrains/compose-multiplatform-core/pull/1810) (aka `BasicTextField2`)
- [Radio buttons and any other elements using `Modifier.selectable` with `Role.RadioButton` will have their state reported to accessibility via `AccessibleValue.getCurrentAccessibleValue()` and as `AccessibleState.CHECKED` in `getAccessibleStateSet()`](https://github.com/JetBrains/compose-multiplatform-core/pull/1797)

### Web

> - [The BasicTextField2 handles browser copy/cut/paste events correctly now](https://github.com/JetBrains/compose-multiplatform-core/pull/1795). Previously, they were ignored

Reviewer 2: There is no `BasicTextField2` name in public API since 1.7
```suggestion
- [The `BasicTextField` handles browser copy/cut/paste events correctly now](https://github.com/JetBrains/compose-multiplatform-core/pull/1795). Previously, they were ignored
```

### iOS

- [Support VoiceControl on iOS](https://github.com/JetBrains/compose-multiplatform-core/pull/1780)
- [`AccessibilitySyncOptions` removed](https://github.com/JetBrains/compose-multiplatform-core/pull/1780). The accessibility tree is built on demand
- [Calculate the order and location of semantic elements in the same way as it's done on Android](https://github.com/JetBrains/compose-multiplatform-core/pull/1809)
- [Support `UIAccessibilityContainerTypeSemanticGroup` for traversal groups](https://github.com/JetBrains/compose-multiplatform-core/pull/1809)
- [Compose works correctly with nested `UIScrollView`s, as well as within `UIScrollView`s](https://github.com/JetBrains/compose-multiplatform-core/pull/1818)
- [Added the ability to close modal Compose view controllers](https://github.com/JetBrains/compose-multiplatform-core/pull/1818) (with non-scrollable content on them) with a swipe gesture

## Fixes

> ### Tests

Reviewer 3: Let's move it to `Multiple Platforms` or discuss adding a new `Tests` category.

(the list categories is strict, the script should notify about that).

The current categories are:
```
    "Multiple Platforms", (implicit Compose Multiple Platforms)
    "iOS", (implicit Compose iOS)
    "Desktop", (implicit Compose Desktop)
    "Web",  (implicit Compose Web)
    "Android",  (implicit Compose Android)
    "Resources",
    "Gradle Plugin",
    "Lifecycle",
    "Navigation",
```
If we decide to create "Test", we would have to create "Interop", "Accessibility", etc, and then somehow divide them by platforms (the current way won't work probably, maybe we should use tags)
